demos/app/iot_demo.c

```diff
@@ -118,19 +118,55 @@ int main( int argc,
     /* For a secured connection, set the members of the credentials. */
     if( demoArguments.securedConnection == true )
     {
-        /* Set credential paths. */
+        /* Set credential information. */
         credentials.pClientCert = demoArguments.pClientCertPath;
```

Comment:
If `demoArguments.pUserName` and `demoArguments.pPassword` are non-NULL and the broker supports authenticating with a username and password, can these cert/private key/root CA fields be set to `NULL`?

Reply:
Yes. They don't have to be, though. That is, enforcement of client certificate authentication over TLS, and username/password authentication at the MQTT broker, are orthogonal.